services/src/main/java/org/keycloak/services/resources/admin/ClientScopeResource.java

```diff
@@ -101,10 +101,11 @@ public Response update(final ClientScopeRepresentation rep) {
 
         try {
             RepresentationToModel.updateClientScope(rep, clientScope);
+            adminEvent.operation(OperationType.UPDATE).resourcePath(session.getContext().getUri()).representation(rep).success();
```

Comment:
out of curiosity: can you please explain to me why we need to send admin event before commiting the transaction?

Reply:
I changed whole event processing a little, see javadoc in [EventListenerProvider](https://github.com/keycloak/keycloak/pull/7645/files#diff-0f446f0779afee5927b69598da2aa364be4c7a01b1932636615d8afca8aec33bR26-R43).

TLDR: All events are now fired only after successful transaction not in the moment when `success()` function is called like it was before.

I am not sure what was the exact reason her for moving it before commit, but it makes sense since we send all events in afterTransaction.